scripts/dat_generate/dat_generate.gml

```diff
@@ -20,14 +21,72 @@ for (a = 0; a <= o.enda; a ++) {
 					if s = 100 blockposition=0
 					if s < 100 blockposition=((s-100)*-1)/100
 					
-					//Add command to result
-					if(o.dat_enableradius) str += "execute at @s[scores={"+objective+"="+string(tick)+"}] run playsound "+ instrument +" "+source+" @a ~ ~ ~ " + string(o.dat_radiusvalue) + " " + string(pitch) + br 
-					else str += "playsound "+ instrument +" "+source+" @s[scores={"+objective+"="+string(tick)+"}] ^" + string(blockposition*2) + " ^ ^ "+string(blockvolume)+ " " + string(pitch) + " 1" + br	
+					// Append -1 or 1 to sound event if note is out of range
+					soundname = instrument
+					if (o.song_key[a, b] <= 32) soundname += "_-1"
+					else if (o.song_key[a, b] >= 58) soundname += "_1"
+					
+					// Add command to result
+					if(o.dat_enableradius) str += "execute at @s run playsound "+ soundname +" "+source+" @a ~ ~ ~ " + string(o.dat_radiusvalue) + " " + string(pitch) + br 
+					else str += "playsound "+ soundname +" "+source+" @s ^" + string(blockposition*2) + " ^ ^ "+string(blockvolume)+ " " + string(pitch) + " 1" + br	

```

Comment:
It appears Minecraft determines the volume based on the camera position, compare:

1. `/execute anchored eyes run playsound minecraft:block.note_block.banjo ambient @s ^ ^ ^ 0.25`
2. `/playsound minecraft:block.note_block.banjo ambient @s ~ ~ ~ 0.25`

Though the difference might be negligible (if there is actually a difference).